WordPress/gutenberg
# Cover block: add option to set opacity when background colour used

## Description
Currently it is not possible to specify opacity on a cover block when a background colour is selected, which prevents the use of some interesting design patterns [as noted here](https://github.com/WordPress/gutenberg/issues/32339).

This PR is experimenting with a slightly different approach to the one taken at https://github.com/WordPress/gutenberg/pull/33927, as there were issues with that one with using a `style="opacity...` due to kses limitations.

This uses the [approach suggest by](https://github.com/WordPress/gutenberg/issues/32339#issuecomment-858137187)
@justintadlock of using the `span` already used by the gradient setting and keeping the structure more consistent.

Fixes: https://github.com/WordPress/gutenberg/issues/32339

## Testing

- Check out PR to local dev env
- Add a cover block with an image background. Within this block nest another Cover block and set a solid background or gradient, and check that you can alter the opacity to show the underlying image without affecting the opacity of the nested blocks children, in the editor and the front end.
- Try loading an old cover block and make sure it migrates ok. If the old block did not have an image it should migrate default dimRatio from 50 -> 100
- Check that the js unit tests run successfully as block fixtures and deprecations updated.

## Screenshots

![colour](https://user-images.githubusercontent.com/3629020/134442543-1c4a900a-7bb8-420d-b3ef-86ccf7720aec.gif)

![gradient](https://user-images.githubusercontent.com/3629020/134442555-eb705040-2413-4f21-8b6f-cf9d5dab2130.gif)
